mdmix/core/models.py

```diff
@@ -0,0 +1,244 @@
+from pathlib import Path
+from typing import List, Dict, Optional, Any, Union
+from pydantic import BaseModel, Field, field_validator
+from enum import Enum
+from datetime import datetime
+
+
+# Enums for validation
+class RestraintMode(str, Enum):
+    FREE = "FREE"
+    HA = "HA"  # Heavy atoms
+    BB = "BB"  # Backbone atoms
+
+
+class EnsembleType(str, Enum):
+    NVT = "NVT"
+    NPT = "NPT"
+
+
+class SimulationEngine(str, Enum):
+    AMBER = "AMBER"
+    NAMD = "NAMD"
+    OPENMM = "OPENMM"
+
+
+class QueueSystem(str, Enum):
+    SLURM = "SLURM"
+    SGE = "SGE"
+    LOCAL = "LOCAL"
+    CTE = "CTE"
+    DAINT = "DAINT"
+
+
+# Request Models (for CLI input)
+class SystemRequest(BaseModel):
+    """Request model for system setup"""
+    name: str = Field(..., description="System name")
+    structure_file: Path = Field(..., description="Path to OFF or PDB file")
+    structure_type: str = Field("off", description="Structure file type (off/pdb)")
+    force_fields: List[str] = Field(
```

Comment:
since some time ago, we don't need to import `typing.List` anymore,
we can go with just `list` (same thing with many other collections and types)
```suggestion
    force_fields: list[str] = Field(
```